src/formatters/utils/sortResults.ts

```diff
@@ -1,13 +1,79 @@
 import { IRuleResult } from '../../types';
 
-export const sortResults = (results: IRuleResult[]) => {
-  return [...results].sort((resultA, resultB) => {
-    const diff = resultA.range.start.line - resultB.range.start.line;
+const compareCode = (one: string | number | undefined, another: string | number | undefined): number => {
+  if (one === void 0 && another === void 0) {
+    return 0;
+  }
+
+  if (one === void 0) {
+    return -1;
+  }
+
+  if (another === void 0) {
+    return 1;
+  }
+
+  return one.toString().localeCompare(another.toString(), undefined, { numeric: true });
```

Comment:
```suggestion
  return String(one).localeCompare(String(another), void 0, { numeric: true });
```
String is a slightly safer alternative. `String.prototype.toString` or `Number.prototype.toString` can always be monkey-patched in JS :mask:

Reply:
Fixed